components/script/bluetooth_blacklist.rs

```diff
@@ -0,0 +1,123 @@
+        }
+    }
+
+    // https://webbluetoothcg.github.io/web-bluetooth/#blacklisted-for-reads
+    pub fn is_blacklisted_for_reads(&self, uuid: &str) -> bool {
+        match self.0 {
+            Some(ref map) => map.get(uuid).map_or(false, |et| et.eq(&ExcludeType::Exclude) ||
+                                                              et.eq(&ExcludeType::ExcludeReads)),
+            None => false,
+        }
+    }
+
+    // https://webbluetoothcg.github.io/web-bluetooth/#blacklisted-for-writes
+    pub fn is_blacklisted_for_writes(&self, uuid: &str) -> bool {
+        match self.0 {
+            Some(ref map) => map.get(uuid).map_or(false, |et| et.eq(&ExcludeType::Exclude) ||
+                                                              et.eq(&ExcludeType::ExcludeWrites)),
+            None => false,
+        }
+    }
+}
+
+// https://webbluetoothcg.github.io/web-bluetooth/#parsing-the-blacklist
+fn parse_blacklist() -> Option<HashMap<String, ExcludeType>> {
+    // Step 1 missing, currently we parse ./resources/gatt_blacklist.txt.
+    let file = match File::open(FILE_PATH) {
+        Ok(file) => file,
+        Err(_) => return None,
+    };
+    let content = BufReader::new(file);
+    // Step 3
+    let mut result = HashMap::new();
+    // Step 2 and 4
+    for line in content.lines() {
+        if let Ok(line) = line {
+            // Step 4.1
+            if !line.is_empty() && !line.starts_with('#') {
+                let mut exclude_type = ExcludeType::Exclude;
+                let mut words = line.split_whitespace();
+                if let Some(uuid) = words.next() {
+                    if validate_uuid(uuid) {
+                        match words.next() {
+                            // Step 4.2 We already have an initialized exclude_type with ExcludeType::Exclude.
+                            None => {},
+                            // Step 4.3
+                            Some(EXCLUDE_READS) => {
+                                exclude_type = ExcludeType::ExcludeReads;
+                            },
+                            Some(EXCLUDE_WRITES) => {
+                                exclude_type  = ExcludeType::ExcludeWrites;
+                            },
+                            // Step 4.4
+                            _ => {
+                                return None;
+                            },
+                        }
+                    } else {
+                        return None;
+                    }
+                    // Step 4.5
+                    if result.contains_key(uuid) {
+                        return None;
+                    }
+                    // Step 4.6
+                    result.insert(uuid.to_string(), exclude_type);
+                }
+            }
+        } else {
+            return None;
+        }
+    }
+    // Step 5
+    return Some(result);
+}
+
+fn validate_uuid(uuid: &str) -> bool {
+    Regex::new(VALID_UUID_REGEX).unwrap().is_match(uuid)
```

Comment:
With this, we create a new `regex` in every line read... we should save the `regex` in a `var` in `parse_blacklist` and use that with `is_match`